controller_manager/src/controller_manager.cpp

```diff
@@ -1263,6 +1274,41 @@ void ControllerManager::list_controllers_srv_cb(
         cs.required_state_interfaces = state_interface_config.names;
       }
     }
+    // check for chained interfaces
+    for (const auto & interface : cs.required_command_interfaces)
+    {
+      auto index = interface.find('/');
+      auto prefix = interface.substr(0, index);
+      if (controller_chain_map.find(prefix) != controller_chain_map.end())
+      {
+        controller_chain_map[cs.name].insert(prefix);
+        controller_chain_interface_map[cs.name].push_back(interface.substr(index+1, interface.size()-1));
+      }
+    }
+    auto references = controllers[i].c->export_reference_interfaces();
+    cs.reference_interfaces.reserve(references.size());
+    for (const auto& reference: references){
+      cs.reference_interfaces.push_back(reference.get_interface_name());
+    }
+    response->controller.push_back(cs);
+    // keep track of controllers that are part of a chain
+    if (!controller_chain_interface_map[cs.name].empty() || controllers[i].c->is_chainable()){
+      chained_controllers.push_back(&response->controller.back());
+    }
+  }
+
+  // create chain connections for all controllers in a chain
+  for (auto & cs : chained_controllers)
```

Comment:
what does `cs` stand for? how about a better name? :D

Reply:
renamed all occurrences of `cs` to `controller_state`